Adds the Release Process for Ansible Community Execution Environment (#70)
ansible-community/images: docs/release_process.md

> ## Release Cadence

Reviewer: ```suggestion
# Releasing the community-ee-* execution environments

## Release Cadence
```

> Community EE is to be built and published on the next day and/or the day after ansible-core is released.

Reviewer: ```suggestion
The community EEs are to be built and published on the next day and/or the day after ansible-core is released.
```

## EE tag versioning

Versioning would be the core tag + a patch number, e.g

    - EE with core `2.16.2` comes out -> `community-ee:2.16.2-1`
    - EE with core `2.16.3` comes out -> `community-ee:2.16.3-1`

## Dependencies

-   Podman/Docker

##  Credentials

- Access to  https://github.com/ansible-community/images repo
- Access to ghcr.io

## Build steps for Community-ee-base

1. Go to the working directory.

`cd images/execution-environments/community-ee-base`

2. Create a new release branch

`git checkout -b base-<ansible-core-version-base-ee-version>`

> 3. Edit the `execution-environment.yml` file to the right `collection version` and `ansible-core version`. Get the correct `[ansible-core](https://pypi.org/project/ansible-core/)` version and `[ansible-<version>.deps](https://github.com/ansible-community/ansible-build-data/blob/main/9/ansible-9.0.1.deps)` and edit accordinly.

Reviewer: Why should the `.deps` file be used here, instead of taking the latest versions of these collections from Ansible Galaxy?

Also if you put links in backticks, they won't work as links and will not be clickable, but show as code. Is that really intended?

Author: > Why should the `.deps` file be used here, instead of taking the latest versions of these collections from Ansible Galaxy?

To ensure collection versions are the correct ones which we know for sure that works. For example when we are releasing 2 versions of community ee 2.15 and 2.16 at the same time.

> Also if you put links in backticks, they won't work as links and will not be clickable, but show as code. Is that really intended?

No,  typo fixing it now.

`vim execution-environments/community-ee-base/execution-environment.yml`

```
version: <base-community-ee with the specific ansible core>
images:
  base_image:
    name: quay.io/fedora/fedora:latest
dependencies:
  ansible_core:

>     package_pip: ansible-core==<ansible-core-version>

Reviewer: There's still my comment from last time:

> One thing that I find strange is that the `release_process.md` document tells the RM to **replace** the EE definition with the content described in this file, instead of simply updating the relevant numbers. Shouldn't the main source of truth be the EE definitions included in this repository?

  ansible_runner:
    package_pip: ansible-runner
  system:
    - openssh-clients
    - sshpass
  galaxy:
    collections:
    - name: ansible.posix
      version: <correct-version>
    - name: ansible.utils
      version: <correct-version>
    - name: ansible.windows
      version: <correct-version>

```

4. List images in local storage

`podman images`

5. Delete the existing community-ee base and fedora images

```
podman rmi <podman_community-ee-base_IMAGE_ID>
podman rmi <podman_community-ee-fedora_IMAGE_ID>
```
6. Create a virtual environment

```
python3 -m venv .venv
source .venv/bin/activate
```
7. Install `ansible-builder` and `setuptools`

```
python3 -m pip install ansible-builder
python3 -m pip install setuptools
```

8. Pull the latest fedora image

`podman pull fedora:latest`

> 9. Build the ansible community-ee-base podman image with ansible-builder. (versioning : 2.16.1-1)

Reviewer: ```suggestion
9. Build the ansible community-ee-base podman image with ansible-builder. (versioning: 2.16.1-1)
```

`ansible-builder build --tag ghcr.io/ansible-community/community-ee-base:<ansible-core-version-base-ee-version>`

> 10. Check if the image has been created or not and get the < image ID> of `community-ee-base`

Reviewer: ```suggestion
10. Check if the image has been created or not and get the <image ID> of `community-ee-base`
```

Copy the token from Github UI and then pass the token on the following command.

13. Login to Github Registry

`echo TOKEN | podman login ghcr.io -u USERNAME --password-stdin`

14. Push both the images (general versioning and the latest) to Github Registry

```
podman push ghcr.io/ansible-community/community-ee-base:<ansible-core-version-base-ee-version>
podman push ghcr.io/ansible-community/community-ee-base:latest
```

15. Commit and push the changes made in the `execution-environments/community-ee-base/execution-environment.yml` to the `https://github.com/ansible-community/images` repo.

```
git add execution-environments/community-ee-base/execution-environment.yml
git commit
git push origin <ansible-core-version-base-ee-version>
```
Compare and create the pull request.

16.   Check the community-ee-base images [here](https://github.com/orgs/ansible-community/packages/container/package/community-ee-base)  and get the sha256 sum.

## Build steps for Community-ee-minimal

1. Go to the working directory

`cd images/execution-environments/community-ee-minimal`

2. Create a new release branch

`git checkout -b base-<ansible-core-version-minimal-ee-version>`

> 3. Edit the `execution-environment.yml` file to the right `collection version` and `ansible-core version`. Get the correct `[ansible-core](https://pypi.org/project/ansible-core/)` version and edit accordinly.

Reviewer: Same here: this isn't a link, but shows the MarkDown link as code. This should rather be:
```suggestion
3. Edit the `execution-environment.yml` file to the right `collection version` and `ansible-core version`. Get the correct [ansible-core](https://pypi.org/project/ansible-core/) version and edit accordinly.
```
